src/pages/ConvertVrt/Convert/index.tsx

```diff
@@ -1,141 +1,218 @@
-  xvsTotalWei,
-  xvsToVrtRate,
-  vrtLimitUsedWei,
-  vrtLimitWei,
+const Convert: React.FC<IConvertProps> = ({
+  xvsToVrtConversionRatio,
   vrtConversionEndTime,
+  userVrtBalanceWei,
+  vrtConversionLoading,
+  userVrtEnabled,
+  convertVrt,
   walletConnected,
 }) => {
   const styles = useStyles();
   const { t, Trans } = useTranslation();
+  const { openSuccessfulTransactionModal } = useSuccessfulTransactionModal();
+  const vrtConverterProxyAddress = getContractAddress('vrtConverterProxy');
+
   const readableXvsAvailable = useConvertToReadableCoinString({
-    valueWei: xvsTotalWei,
+    valueWei: userVrtBalanceWei?.times(xvsToVrtConversionRatio || 1),
     tokenId: XVS_ID,
   });
+  const readableUserVrtBalance = useConvertToReadableCoinString({
+    valueWei: userVrtBalanceWei,
+    tokenId: VRT_ID,
+  });
+
+  const calculateXvsFromVrt = useCallback(
+    (value: BigNumber) =>
+      !value.isNaN() && !value.isZero() && xvsToVrtConversionRatio
+        ? new BigNumber(value).times(xvsToVrtConversionRatio)
+        : undefined,
+    [xvsToVrtConversionRatio],
+  );
+
   const { relativeTimeTranslationKey, realtiveTimeFormatValues } =
     formatI18nextRelativetimeValues(vrtConversionEndTime);
-
+  const onSubmit = async (vrtAmount: string) => {
+    try {
+      const vrtAmountWei = convertCoinsToWei({ value: new BigNumber(vrtAmount), tokenId: VRT_ID });
+      const transactionHash = await convertVrt(vrtAmountWei.toFixed());
+      // Display successful transaction modal
+      if (!xvsToVrtConversionRatio) {
+        // This should never happen because the form is not rendered without successfully fetching this
+        throw new InternalError(t('convertVrt.internalErrorXvsToVrtConversionRatioUndefined'));
+      }
+      const xvsAmountWei = calculateXvsFromVrt(vrtAmountWei);
+      openSuccessfulTransactionModal({
+        title: t('convertVrt.successfulConvertTransactionModal.title'),
+        transactionHash,
+        children: (
+          <div css={styles.successModalConversionAmounts}>
+            <Token symbol={VRT_ID} css={styles.successModalToken} variant="small2" />
+            <Typography variant="small2" css={[styles.fontWeight600, styles.successMessage]}>
+              {convertWeiToCoins({
+                valueWei: vrtAmountWei,
+                tokenId: VRT_ID,
+                returnInReadableFormat: true,
+              })}
+            </Typography>
+            <Icon name="arrowShaft" css={styles.successModalArrow} />
+            <Token symbol={XVS_ID} css={styles.successModalToken} variant="small2" />{' '}
+            <Typography variant="small2" css={[styles.fontWeight600, styles.successMessage]}>
+              {xvsAmountWei &&
+                convertWeiToCoins({
+                  valueWei: xvsAmountWei,
+                  tokenId: XVS_ID,
+                  returnInReadableFormat: true,
+                })}
+            </Typography>
+          </div>
+        ),
```

Comment:
This will create some conflicts if `message` and `amount` are also passed to the function.
How about renaming `message` to `content` inside `SuccessfulTransactionModal` and make it so it can be a string or a React element?
We'd be able to do what you're doing here and avoid potential visual glitches from passing too many props.